internal/broker/instance_create.go

```diff
@@ -359,6 +373,31 @@ func (b *ProvisionEndpoint) validate(ctx context.Context, details domain.Provisi
 		)
 	}
 
+	var awsClient aws.Client
+	if b.providerSpec.ZonesDiscovery(pkg.CloudProviderFromString(values.ProviderType)) {
```

Comment:
if AZure/GCP discovery zones are enabled (for example by accident) we have no implementation for it - all provisioning fails, let's discuss